docs/index.bs

```diff
@@ -3177,6 +3231,41 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. Return |response|.
   </section>
 
+  <section algorithm>
+    <h3 id="verify-router-rule-algorithm"><dfn>VerifyRouterRule</dfn></h3>
+
+      : Input
+      :: |rule|, a {{RouterRule}}
+      : Output
+      :: a boolean
+
+      1. If |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"] is the empty string, return false.
+      1. Let |urlPattern| be a result of [=urlpattern/parsing=] |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]. If a <code>TypeError</code> is thrown, return false.
+      1. [=list/For each=] |part| of |urlPattern|:
+          1. If |part|'s [=urlpattern-part/type=] is "regexp", return false.
+
+              Note: Since running a user-defined regular expression has a security concern, it is prohibited.
+
+      1. Set |urlPattern| to |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]'s associated [=hostname component=].
+      1. Return true.
+  </section>
```

Comment:
Probably we should return false if the pattern has any nontrivial components besides hostname, right? Otherwise people might try to use patterns like `/:foo` and not realize that will do nothing.

(I don't know how to do that precisely since, per the above, I don't know how to get from a list of parts and a string to the relevant components.)

Reply:
If it is outside of acceptable patterns, I believe TypeError would be raised by https://wicg.github.io/urlpattern/#parse-a-pattern-string.  If not, since there should be components, and it might be fine to use it for pattern matching.

By the way, if you are talking about supporting URLPattern "field"? other than the pathname, I recently lean on using URLPatternInit instead of USVString.  I mean that if developers want to do something complicated, they should use URLPatternInit.  USVString is only usable for pathnames.  What do you think?

Reply:
Upon your suggestion offline, I have updated the specification to support a full set of URLPattern instead of only pathname.  I suppose the comment became obsolete now.